server/test/unit/server/controllers/test_repository.py

```diff
@@ -0,0 +1,161 @@
+        u_fields = ['key_field']
+        mock_get_model.return_value = DemoModel
+        mock_demo_objects.return_value.only.return_value = [test_unit]
+        result = list(repo_controller.find_repo_content_units(repo, units_q=u_filter,
+                                                              units_fields=u_fields))
+
+        mock_demo_objects.return_value.only.assert_called_once_with(['key_field'])
+
+        # validate that the repo content unit was returned and that the unit is attached
+        self.assertEquals(result, [test_rcu])
+        self.assertEquals(result[0].unit, test_unit)
+
+    @patch.object(DemoModel, 'objects')
+    @patch('pulp.server.controllers.repository.plugin_api.get_unit_model_by_id')
+    def test_limit(self, mock_get_model, mock_demo_objects, mock_rcu_objects):
+        """
+        Test that limits are applied properly to the results
+        """
+        repo = MagicMock(repo_id='foo')
+        rcu_list = []
+        unit_list = []
+        for i in range(10):
+            unit_id = 'bar_%i' % i
+            unit_key = 'key_%i' % i
+            rcu = model.RepositoryContentUnit(repo_id='foo',
+                                              unit_type_id='demo_model',
+                                              unit_id=unit_id)
+            rcu_list.append(rcu)
+            unit_list.append(DemoModel(id=unit_id, key_field=unit_key))
+
+        mock_rcu_objects.return_value = rcu_list
+
+        mock_get_model.return_value = DemoModel
+        mock_demo_objects.return_value = unit_list
+        result = list(repo_controller.find_repo_content_units(repo, limit=5))
+
+        self.assertEquals(5, len(result))
+        self.assertEquals(result[0].unit_id, 'bar_0')
+        self.assertEquals(result[4].unit_id, 'bar_4')
+
+    @patch.object(DemoModel, 'objects')
+    @patch('pulp.server.controllers.repository.plugin_api.get_unit_model_by_id')
+    def test_skip(self, mock_get_model, mock_demo_objects, mock_rcu_objects):
+        """
+        Test that the skip parameter is applied properly
+        """
+        repo = MagicMock(repo_id='foo')
+        rcu_list = []
+        unit_list = []
+        for i in range(10):
+            unit_id = 'bar_%i' % i
+            unit_key = 'key_%i' % i
+            rcu = model.RepositoryContentUnit(repo_id='foo',
+                                              unit_type_id='demo_model',
+                                              unit_id=unit_id)
+            rcu_list.append(rcu)
+            unit_list.append(DemoModel(id=unit_id, key_field=unit_key))
+
+        mock_rcu_objects.return_value = rcu_list
+
+        mock_get_model.return_value = DemoModel
+        mock_demo_objects.return_value = unit_list
+        result = list(repo_controller.find_repo_content_units(repo, limit=5, skip=5))
+
+        self.assertEquals(5, len(result))
+        self.assertEquals(result[0].unit_id, 'bar_5')
+        self.assertEquals(result[4].unit_id, 'bar_9')
+
```

Comment:
Not a requirement but consider adding a test that uses both skip and limit together.